components/org.wso2.carbon.identity.application.authenticator.biometric/src/main/resources/Artifacts/wait.jsp

```diff
@@ -0,0 +1,135 @@
+    <meta charset="utf-8"/>
+
+    <link rel="icon" href="images/favicon.png" type="image/x-icon"/>
+    <link href="libs/bootstrap_3.3.5/css/bootstrap.min.css" rel="stylesheet">
+    <link href="css/Roboto.css" rel="stylesheet">
+    <link href="css/custom-common.css" rel="stylesheet">
+    <link href="css/loading1.css" rel="stylesheet">
+
+    <script language="JavaScript" type="text/javascript" src="libs/jquery_1.11.3/jquery-1.11.3.js"></script>
+    <script language="JavaScript" type="text/javascript" src="libs/bootstrap_3.3.5/js/bootstrap.min.js"></script>
+
+    <header class="header header-default">
+        <div class="container-fluid"><br></div>
+        <div class="container-fluid">
+            <div class="pull-left brand float-remove-xs text-center-xs">
+                <a href="#">
+                    <img src="images/logo-inverse.svg" alt="WSO2" title="WSO2" class="logo">
+                    <h1><em>Identity Server</em></h1>
+                </a>
+            </div>
+        </div>
+    </header>
+</head>
+<body>
+<h2>Please check your mobile device and authenticate with the fingerprint</h2>
+<div class="loader"></div>
+<form id="toCommonAuth" action="<%=commonauthURL%>" method="POST" style="display:none;">
+    <label for="sessionDataKey">sessionDataKey</label><input id="sessionDataKey" name="sessionDataKey">
+    <label for="signedChallenge">signedChallenge</label><input id="signedChallenge" name="signedChallenge">
+</form>
+</body>
+
+<%--// TODO: make a war file without putting to auth endpoint-Discuss with Pulasthi ayya.
+ --%>
+<script type="text/javascript">
+
+    let sessionDataKey;
+    let signedChallenge;
+    const refreshInterval = 1000;
+    const timeout = 20000;
+    let biometricEndpointWithQueryParams = "<%=BiometricAuthenticatorConstants.DOMAIN_NAME + BiometricAuthenticatorConstants.BIOMETRIC_ENDPOINT + BiometricAuthenticatorConstants.POLLING_QUERY_PARAMS%>";
+    const GET = 'GET';
+
+    $(document).ready(function () {
+        var startTime = new Date().getTime();
+        console.log("Start time: "+ startTime);
+
+        const intervalListener = window.setInterval(function () {
+            checkWaitStatus();
+        }, refreshInterval);
+
+        function checkWaitStatus() {
+            var now = new Date().getTime();
+            if ((startTime + timeout) < now) {
+                alert("timeout triggered");
+                window.clearInterval(intervalListener);
+                window.location.replace("retry.do");
+            }
+
+            const urlParams = new URLSearchParams(window.location.search);
+            sessionDataKey = urlParams.get('sessionDataKey');
+
+            $.ajax(biometricEndpointWithQueryParams + sessionDataKey, {
+                async: false,
+                method: GET,
+                // todo: check whether there are any problems when same get req is sent over n over again, will there be cache issues? any solutions?-Future Improvement-Include in DOCs
```

Comment:
What is the status of this TODO?

Reply:
No cache issues were encountered when i tried sending GET requests from 2 Dispatch service providers. both push notifications are queried in the tray and can be authenticated separately.